okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -514,6 +515,18 @@ public void proxy(RoutingContext ctx) {
       return; // Error code already set in ctx
     }
 
+    FolioLoggingContext.put("tenantId", tenantId);
+
+    if (ctx.request() != null && ctx.request().headers() != null) {
+
+      FolioLoggingContext.put("requestId", ctx.request().headers().get(XOkapiHeaders.REQUEST_ID));
+
+      FolioLoggingContext.put("moduleId", ctx.request().headers().get(XOkapiHeaders.MODULE_ID));
+
+      FolioLoggingContext.put("userId", ctx.request().headers().get(XOkapiHeaders.USER_ID));
+
+    }
+
     final MultiMap headers = ctx.request().headers();
     sanitizeAuthHeaders(headers);
     tenantManager.get(tenantId, gres -> {
```

Comment:
Redundant checks here.. headers are always present.  Move your stuff below the headers local and have it more readable, please.

Reply:
done